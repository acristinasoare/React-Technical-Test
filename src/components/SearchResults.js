import React from "react";
import "../styles/searchResults.css";

function SearchResults({ results }) {
  if (!results.length) {
    return <p className="no-results">No results</p>;
  }
  return (
    <>
      <div className="grid-container">
        {results.map((image) => (
          <img key={image} className="grid-item" src={image} alt="spaceImage" />
        ))}
      </div>
    </>
  );
}

export default SearchResults;
