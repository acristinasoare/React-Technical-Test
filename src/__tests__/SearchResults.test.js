import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  it("renders 'No results' when results are not provided", () => {
    const results = [];
    const { asFragment } = render(<SearchResults results={results} />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("No results")).toBeInTheDocument();
  });
  it("renders the image with correct src and alt attributes", () => {
    const results = ["image1.jpg", "image2.jpg", "image3.jpg"];
    render(<SearchResults results={results} />);
    expect(screen.queryByText("No results")).not.toBeInTheDocument();
    const images = screen.queryAllByAltText("spaceImage");
    expect(images).toHaveLength(results.length);
  });
});
