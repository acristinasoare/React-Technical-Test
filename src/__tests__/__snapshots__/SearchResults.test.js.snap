// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders 'No results' when results are not provided 1`] = `
<DocumentFragment>
  <p
    class="no-results"
  >
    No results
  </p>
</DocumentFragment>
`;
